pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -457,26 +457,34 @@ func makeUID() string {
 // getTerminationMessage looks on the filesystem for the provided termination message path, returning a limited
 // amount of those bytes, or returns true if the logs should be checked.
 func getTerminationMessage(status *runtimeapi.ContainerStatus, terminationMessagePath string, fallbackToLogs bool) (string, bool) {
-	if len(terminationMessagePath) == 0 {
+	path, found := getTerminationMessagePath(status, terminationMessagePath)
+	if !found {
 		return "", fallbackToLogs
 	}
+
+	data, _, err := tail.ReadAtMost(path, kubecontainer.MaxContainerTerminationMessageLength)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return "", fallbackToLogs
+		}
+		return fmt.Sprintf("Error on reading termination log %s: %v", path, err), false
+	}
+	return string(data), fallbackToLogs && len(data) == 0
```

Comment:
I don't understand this logic, the `fallbackToLogs` is never assigned any value so it will be `false`. So this condition is always `false`. What am I missing?

Reply:
it was the same code before... maybe I am missing something

Reply:
The fallbackToLogs is set by the caller function convertToKubeContainerStatus().
```go
func (m *kubeGenericRuntimeManager) convertToKubeContainerStatus(status *runtimeapi.ContainerStatus) (cStatus *kubecontainer.Status) {
    cStatus = toKubeContainerStatus(status, m.runtimeName)
    if status.State == runtimeapi.ContainerState_CONTAINER_EXITED {
        // Populate the termination message if needed.
        annotatedInfo := getContainerInfoFromAnnotations(status.Annotations)
        // If a container cannot even be started, it certainly does not have logs, so no need to fallbackToLogs.
        fallbackToLogs := annotatedInfo.TerminationMessagePolicy == v1.TerminationMessageFallbackToLogsOnError &&
            cStatus.ExitCode != 0 && cStatus.Reason != "ContainerCannotRun"
        tMessage, checkLogs := getTerminationMessage(status, annotatedInfo.TerminationMessagePath, fallbackToLogs)
```
I think that the fallbackToLogs would be true for a pod like below.
```
apiVersion: v1
kind: Pod
metadata:
  name: termination-demo
spec:
  containers:
  - name: termination-demo-container
    image: debian
    imagePullPolicy: IfNotPresent
    terminationMessagePolicy: FallbackToLogsOnError
    command: ["/bin/sh"]
    args: ["-c", "sleep 10 && echo Sleep expired > /dev/termination-log && exit 2"]
```